rest-service/manager_rest/storage/resource_models.py

```diff
@@ -2038,6 +2046,43 @@ def set_node(self, node):
     def allowed_filter_attrs(cls):
         return ['id']
 
+    @property
+    def is_status_check_ok(self):
+        """Has the last status check for this NI succeeded?
+
+        This examines the result of the most recent check_status call
+        on this node instance, and returns whether the call succeeded.
+
+        If the result is missing, the result is succeeded.
+        """
+        props = self.system_properties or {}
+        status = props.get('status') or {}
+        return bool(status.get('ok', True))
+
+    @property
+    def has_configuration_drift(self):
+        """Has this NI's configuration drifted?
+
+        This examines the result of the most recent check_drift call
+        on this node instance, and returns whether there was any configuration
+        drift reported.
+
+        The instance is drifted if either the instance itself, or any of its
+        relationships have drifted.
+        """
+        props = self.system_properties or {}
+        instance_drift = props.get('configuration_drift') or {}
+        sources_drift = \
+            props.get('source_relationships_configuration_drift') or {}
+        targets_drift = \
+            props.get('target_relationships_configuration_drift') or {}
+        for drift in itertools.chain(
+            [instance_drift], sources_drift.values(), targets_drift.values()
+        ):
+            if not drift.get('ok', True) or drift.get('result') is not None:
```

Comment:
When would drift['ok'] be True, but drift['result'] be not null?

Reply:
`ok` is whether the operation succeeded, and result is... whatever the operation returns, which is up to the plugin author

Reply:
So there's the question of how do we want to signal drift, either by raising an exception, or by returning something, and here we allow both...

Reply:
Hmm... I suppose the operation checking the drift could set informational stuff in its logs rather than output. I'm not sure whether this feels right or not, but I guess we can just wait and see how this ends up being used as it could go either way.